### PR TITLE
Manager doc update

### PR DESCRIPTION
**Pull request by brycegbrazen**
Remove temp doc message and link to OpenAssetIO Manager Python and Plugin for Ayon

## Description

- [x] I have updated the release notes.
- [x] I have updated all relevant user documentation.

### CODE REVIEW

File: doc/doxygen/src/Examples.dox [hunk 1]
```diff
@@ -64,7 +64,12 @@
  * that some example @ref asset_management_system has a plugin,
  * installed on @ref plugin_path_var).
  *
- * We will be providing an example manager implementation soon!
+ * Check out <a href="https://github.com/OpenAssetIO/Template-OpenAssetIO-Manager-Python"
```
Comment (brycegbrazen) [inline doc/doxygen/src/Examples.dox:67]: Not sure if this is the best way to reference these links. Also, they are not bold right now so they are hard to see.

Comment (feltech) [reply doc/doxygen/src/Examples.dox:67]: I think that's fine. These are both linked on the home page too ("An Introduction").

Comment (feltech) [inline doc/doxygen/src/Examples.dox:67]: [fix] I'd be happy just keeping the removal and no need to link the Template or Ayon.

If you want to keep these links, can we change `OpenAssetIO Manager Python` to `Python Manager Template`?

**Review by feltech — CHANGES_REQUESTED** [review]

File: doc/doxygen/src/Examples.dox (hunk 1 above)
Comment (brycegbrazen) [reply doc/doxygen/src/Examples.dox:67]: I'm personally partial to adding additional cross-references in documentation so people can more easily find things. Let me know if you disagree though!

Comment (brycegbrazen) [reply doc/doxygen/src/Examples.dox]: Fixed the styling so they are bold 👍

**Review by feltech — APPROVED** [review]

**Comment by feltech** [comment]
Fine for this case, but in general we'd avoid commits that just fix issues with previous commits in the same PR, e.g. in this case 0d2cfb5a5dbee9861adc29410224e468f0f2217e. 

For a small PR like this you could just edit the commit and force push. 

But more generally, and for future reference, we make use of git [autosquash](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt---autosquash),  In this particular example the commit title would be `fixup! [Docs] Improve manager documentation` or `squash! [Docs] Improve manager documentation`. You can then push that commit, and put the commit hash in the PR comment thread, providing a handy link for the reviewer to see how the comment was addressed. When the PR is ready to merge you can then `git rebase -i --autosquash` to get rid of the `fixup!`s, then force-push.  See also [contributing docs](https://github.com/OpenAssetIO/OpenAssetIO/blob/main/doc/contributing/PULL_REQUESTS.md#our-pull-request-process).

**Comment by feltech** [comment]
I've taken the liberty of squashing and rebasing.

**Pull request merged**